src/Features/CSharp/Portable/GoToDefinition/CSharpGoToDefinitionSymbolService.cs

```diff
@@ -20,5 +22,77 @@ public CSharpGoToDefinitionSymbolService()
 
         protected override ISymbol FindRelatedExplicitlyDeclaredSymbol(ISymbol symbol, Compilation compilation)
             => symbol;
+
+        protected override int? GetTargetPositionIfControlFlow(SyntaxToken token)
+        {
+            var node = token.GetRequiredParent();
+
+            switch (token.Kind())
+            {
+                case SyntaxKind.ContinueKeyword:
+
+                    var foundContinuedLoop = TryFindContainingLoop(node);
+
+                    return foundContinuedLoop?.IsContinuableConstruct() == true
+                        ? foundContinuedLoop.GetFirstToken().Span.Start
+                        : null;
+
+                case SyntaxKind.BreakKeyword:
+                    if (token.GetPreviousToken().IsKind(SyntaxKind.YieldKeyword))
+                    {
+                        goto case SyntaxKind.YieldKeyword;
+                    }
+
+                    var foundBrokenLoop = TryFindContainingLoop(node);
+
+                    return foundBrokenLoop?.IsBreakableConstruct() == true
+                        ? foundBrokenLoop.GetLastToken().Span.End
+                        : null;
+
+                case SyntaxKind.YieldKeyword:
+                case SyntaxKind.ReturnKeyword:
+                    {
+                        var foundReturnableConstruct = TryFindContainingReturnableConstruct(node);
+                        return foundReturnableConstruct?.GetFirstToken().Span.Start;
```

Comment:
likely not the ideal.  this will put us on the attributes if they are there.  can you get us the location of the name?  there may be a helper for this.

Reply:
i think the easiest way is to try o get the symbol for the node if it exists, then use symbol.Locations.  if it doesn't ahve any, fallback to just the node you have like you do now.

Reply:
@CyrusNajmabadi I'm okay with placing the cursor elsewhere, but I think doing it syntactically seems easier. 
What's wrong with just recognizing the different kinds of methods, lambdas and local functions?

Reply:
> What's wrong with just recognizing the different kinds of methods, lambdas and local functions?

i'm not strictly opposed to that.  it's more that it means keeping that list maintained.  for example, if we add a new construct to the language.  THe benefit of ISymbol.Location is that it exists for this purpose.

> but I think doing it syntactically seems easier.

I'm not certain.  LMK what you find.  I personally think symbolic would be easier.  but i could be wrong here.  I also don't my symbolic as goto-def is normally semantic work anyways, so it's not like it's a bad thing for us to go to symbols here.